cookbooks/cdo-varnish/libraries/helpers.rb

```diff
@@ -0,0 +1,117 @@
+# Various Ruby helper methods to generate Varnish VCL from the base configuration.
+
+# Basic regex matcher for the optional query part of a URL.
+QUERY_REGEX = "(\\?.*)?$"
+
+def path_to_regex(path)
+  path = "^/#{path.sub(/^\//,'')}"
+  if path[-1] == '*'
```

Comment:
This will generate incorrect/invalid regular expressions  or surprising behavior for a number of plausible path expressions that should be rejected and for which we would want to have unit tests.

Consider for example:
/images/*.jpg    (path with wildcarded filename)
/assets/*/images   (path with wildcard subexpression)
/path.1/2            (regular expression characters with special meaning like ., [, \, etc)
/a???/        (unsupported wildcard character supported in CDN paths)